paramak/shape.py

```diff
@@ -158,6 +158,39 @@ def union(self):
     def union(self, value):
         self._union = value
 
+    @property
+    def largest_dimension(self):
+        """Calculates a bounding box for the Shape and returns the largest
+        absolute value of the largest dimension of the bounding box"""
+        largest_dimension = 0
+        if isinstance(self.solid, cq.Compound):
+            for solid in self.solid.Solids():
+                largest_dimension = max(
+                    abs(self.solid.BoundingBox().xmax),
+                    abs(self.solid.BoundingBox().xmin),
+                    abs(self.solid.BoundingBox().ymax),
+                    abs(self.solid.BoundingBox().ymin),
+                    abs(self.solid.BoundingBox().zmax),
+                    abs(self.solid.BoundingBox().zmin),
+                    largest_dimension
+                )
+        else:
+            largest_dimension = max(
+                abs(self.solid.val().BoundingBox().xmax),
+                abs(self.solid.val().BoundingBox().xmin),
+                abs(self.solid.val().BoundingBox().ymax),
+                abs(self.solid.val().BoundingBox().ymin),
+                abs(self.solid.val().BoundingBox().zmax),
+                abs(self.solid.val().BoundingBox().zmin),
+                largest_dimension
+            )
+        self._largest_dimension = largest_dimension
```

Comment:
```suggestion
        self.largest_dimension = largest_dimension
```
If underscore is used, `@largest_dimension.setter` won't be used.